src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -376,6 +376,18 @@ public function initialized($id)
     }
 
     /**
+     * Clears shared services from the container.
+     */
+    public function clear()
+    {
+        if (!empty($this->scopedServices)) {
+            throw new RuntimeException('Clearing the container is not allowed when a scope is active.');
+        }
+
+        $this->services = array();
```

Comment:
What about aliases and some other properties? I know that this does not help with your goal, but this implementation looks like a hack to me. Or you should call it `clearSharedServices()` or something to really show the intent.